tests/test_control_low.py

```diff
@@ -207,6 +207,67 @@ def test_mqtt(self):
             oldmqtt["keep_alive_interval"],
         )
 
+    # Available from fw version 2.7.1
+    @vcr.use_cassette("tests/cassettes/TestControlInterface.test_color.yaml")
+    def test_color(self):
+        ctr = ControlInterface(self.host)
+
+        res = ctr.set_mode("color")._data
+        if self.isrecording:
+            print("set_mode:", res)
+        else:
+            self.assertEqual(res, {"code": 1000})
+
+        res = ctr.set_led_color_rgb(128, 128, 0)._data
+        if self.isrecording:
+            print("set_led_color_rgb:", res)
+        else:
+            self.assertEqual(res, {"code": 1000})
+
+        res = ctr.get_led_color()._data
+        if self.isrecording:
+            print("get_led_color:", res)
+        else:
+            self.assertEqual(
+                res,
+                {
+                    "hue": 60,
+                    "saturation": 255,
+                    "value": 128,
+                    "red": 128,
+                    "green": 128,
+                    "blue": 0,
+                    "code": 1000,
+                },
+            )
+
+        res = ctr.set_led_color_hsv(180, 255, 255)._data
+        if self.isrecording:
+            print("set_led_color_hsv:", res)
+        else:
+            self.assertEqual(res, {"code": 1000})
```

Comment:
Let's move these up where RGB are set.

Then in both `get_led_color()` and `get_mode()` we could test for these HSV values.

Reply:
But it does set the values before testing, with both tests! 
Note that there is only one color set at any one time on the leds, but this color can be specified *either* through RGB or HSV. So when setting the RGB to (128, 128, 0), i.e. dim yellow, the HSV coordinates are updated as well, to (60, 255, 128).  (Hue 60 means yellow, value 128 means half-strength). These values are then tested in the test. Then the HSV coordinates are set to (180, 255, 255), ie full strength cyan (hue 180 means cyan, value 255 is full strength), which results in the RGB coordinates being correctly updated to (0, 255, 255). Which is again tested, but this time through `get_mode`.

Reply:
> Note that there is only one color set at any one time on the leds, but this color can be specified either through RGB or HSV.

Right, this is what I haven't realized. My bad. I'll explain this commented against line that failed.